hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/test/AssertionSinks.java

```diff
@@ -138,6 +146,46 @@ private AssertionSinks() {
             .build();
     }
 
+    /**
+     * Asserts that the previous stage emitted the exact sequence of
+     * expected items and nothing else.
+     * <p>
+     * Is just a wrapper around {@link #assertCollectedEventually(int, ConsumerEx)},
+     * providing an {@code acceptFn} for doing the order-sensitive check.
+     * Consult that javadoc for further details on how to use.
+     */
+    @Nonnull
+    public static <T> Sink<T> assertOrderedEventually(
+            int timeoutSeconds,
+            @Nonnull Collection<? extends T> expected
+    ) {
+        final List<? super T> exp = new ArrayList<>(expected);
+        return assertCollectedEventually(timeoutSeconds,
+                received -> assertEquals(null, exp, received));
+    }
+
+    /**
+     * Asserts that the previous stage emitted the expected items in any order,
+     * but nothing else.
```

Comment:
How do you check the "nothing else" part?

Reply:
The javadoc is wrong, since you don't. It completes as long as those items are received but there can be subsequent items.